packages/wallet-sdk/src/sign/extension/ExtensionSigner.ts

```diff
@@ -27,14 +29,24 @@ export class ExtensionSigner implements Signer {
     this.extensionProvider = extensionProvider;
 
     this.extensionProvider.on('chainChanged', (chainId) => {
-      this.updateListener.onChainUpdate({ id: Number(chainId) });
+      this.updateListener.onChainIdUpdate(Number(chainId));
     });
 
     this.extensionProvider.on('accountsChanged', (accounts) =>
       this.updateListener.onAccountsUpdate(accounts as AddressString[])
     );
   }
 
+  get accounts() {
+    return this.extensionProvider.send({ method: 'eth_accounts' }) as AddressString[];
+  }
+
+  get chain() {
+    const hexString = this.extensionProvider.send({ method: 'eth_chainId' }) as HexString;
+    // TODO: currently, provider expects `rpcUrl` for fetch requests
+    return { id: intNumberFromHexString(hexString) };
+  }
```

Comment:
yes, this looks ugly. will revisit. ideally, it would be great if we don't need to expose this at all.